server/main.lua

```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
-				sourceplayer = ped.PlayerData.source
-			}
-		end
+	for _, v in pairs(closePlayers) do
+		local player = exports.qbx_core:GetPlayer(v.id)
+		players[#players + 1] = {
+			id = v.id,
+			name = player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname,
+			citizenid = player.PlayerData.citizenid,
+			job = player.PlayerData.job,
+			gang = player.PlayerData.gang,
+			source = player.PlayerData.source
+		}
 	end
 
 	table.sort(players, function(a, b)
 		return a.name < b.name
 	end)
 
 	return players
-end
+end)
+
+---@param accountName string
+---@param group 'job'|'gang'
+---@return table?
+lib.callback.register('qbx_management:server:getemployees', function(source, accountName, group)
+	return getEmployees(source, accountName, group)
+end)
+
+---@param data table
+---@return table?
+RegisterNetEvent('qbx_management:server:updateGrade', function(data)
+	updateGrade(source, data)
+	TriggerClientEvent('qbx_management:client:openMenu', source, data.group)
+end)
 
+---@param data {source: number, grade: number, group: 'job'|'gang'}
+RegisterNetEvent('qbx_management:server:HireEmployee', function(data)
+	local player = exports.qbx_core:GetPlayer(source)
+	local target = exports.qbx_core:GetPlayer(data.source)
+	
+    if not player.PlayerData[data.group].isboss then
+        exploitBan(source, 'HireEmployee Exploiting')
+        return
+    end
+	
+    if not target then
+        exports.qbx_core:Notify(source, 'Civilian is not in city.', 'error')
+        return
+    end
+
+	local jobName = player.PlayerData[data.group].name
+	local logArea = data.group == 'gang' and 'gang' or 'boss'
+
+    local success = data.group == 'gang' and target.Functions.SetGang(jobName, data.grade) or target.Functions.SetJob(jobName, data.grade)
+    local grade = data.group == 'gang' and gangs[jobName].grades[data.grade].name or jobs[jobName].grades[data.grade].name
+
+    if success then
+        local playerFullName = player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname
+        local targetFullName = target.PlayerData.charinfo.firstname .. ' ' .. target.PlayerData.charinfo.lastname
+        local organizationLabel = player.PlayerData[data.group].label
+
+        exports.qbx_core:Notify(source, 'You hired ' .. targetFullName .. ' into ' .. organizationLabel, 'success')
+        exports.qbx_core:Notify(target.PlayerData.source, 'You have been hired into ' .. organizationLabel, 'success')
+        TriggerEvent('qb-log:server:CreateLog', logArea..'menu', grade, 'yellow', playerFullName.. ' successfully recruited ' .. targetFullName .. ' (' .. organizationLabel .. ')', false)
+    else
+        exports.qbx_core:Notify(source, 'Couldn\'t hire civilian', 'error')
+    end
+	TriggerClientEvent('qbx_management:client:openMenu', source, data.group)
+end)
+
+---@param data {source: number, group: 'job'|'gang'}
+RegisterNetEvent('qbx_management:server:fireEmployee', function(data)
```

Comment:
This is a large function. Consider if it can be broken up into smaller functions.